common/test/acceptance/tests/lms/test_learner_profile.py

```diff
@@ -0,0 +1,259 @@
+
+    def visit_my_profile_page(self, user, privacy=None):
+        """
+        Visits a users profile page.
+        """
+        self.authenticate_as_user(user)
+        self.my_profile_page.visit()
+        self.my_profile_page.wait_for_page()
+
+        if user is self.MY_USER and privacy is not None:
+            self.my_profile_page.privacy = privacy
+
+            if privacy == self.PRIVACY_PUBLIC:
+                self.set_pubilc_profile_fields_data(self.my_profile_page)
+
+    def visit_other_profile_page(self, user, privacy=None):
+        """
+        Visits a users profile page.
+        """
+        self.authenticate_as_user(user)
+        self.other_profile_page.visit()
+        self.other_profile_page.wait_for_page()
+
+        if user is self.OTHER_USER and privacy is not None:
+            self.other_profile_page.privacy = privacy
+
+            if privacy == self.PRIVACY_PUBLIC:
+                self.set_pubilc_profile_fields_data(self.other_profile_page)
+
+    def test_dashboard_learner_profile_link(self):
+        """
+        Scenario: Verify that my profile link is present on dashboard page and we can navigate to correct page.
+
+        Given that I am a registered user.
+        When I go to Dashboard page.
+        And I click on username dropdown.
+        Then I see My Profile link in the dropdown menu.
+        When I click on My Profile link.
+        Then I will be navigated to My Profile page.
+        """
+        self.dashboard_page.visit()
+        self.dashboard_page.click_username_dropdown()
+        self.assertTrue('My Profile' in self.dashboard_page.username_dropdown_link_text)
+        self.dashboard_page.click_my_profile_link()
+        self.my_profile_page.wait_for_page()
+
+    def test_fields_on_my_private_profile(self):
+        """
+        Scenario: Verify that desired fields are shown when looking at her own private profile.
+
+        Given that I am a registered user.
+        And I visit My Profile page.
+        And I set the profile visibility to private.
+        And I reload the page.
+        Then I should see the profile visibility selector dropdown.
+        Then I see some of the profile fields are shown.
+        """
+        self.visit_my_profile_page(self.MY_USER, privacy=self.PRIVACY_PRIVATE)
+
+        self.assertTrue(self.my_profile_page.privacy_field_visible)
+        self.assertEqual(self.my_profile_page.visible_fields, self.PRIVATE_PROFILE_FIELDS)
+
+    def test_fields_on_my_public_profile(self):
+        """
+        Scenario: Verify that desired fields are shown when looking at her own public profile.
+
+        Given that I am a registered user.
+        And I visit My Profile page.
+        And I set the profile visibility to public.
+        And I reload the page.
+        Then I should see the profile visibility selector dropdown.
+        Then I see all the profile fields are shown.
+        And `location`, `language` and `about me` fields are editable.
```

Comment:
Please add tests for setting each of these fields that verify that the values are persisted after reloading the page.